### PR TITLE
Review branch

### PR DESCRIPTION
**Pull request by jordaninthewind**
Request for review and merging to master

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,18 +5,27 @@
   "main": "index.js",
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
-    "tasK:2.1": ""
+    "tasK:2.1": "nodemon ./src/index.js --exec babel-node -e js",
```
Comment (MurDaD) [inline package.json]: `task:2.1`

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import express from 'express';
+import router from './routes/routes';
+
+const app = express();
+const PORT = 3000;
```
Comment (MurDaD) [inline src/index.js]: this should be a default port

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+
+app.use('/', router);
+
+app.listen(PORT, () => console.log(`listening on port ${PORT}`));
```
Comment (MurDaD) [inline src/index.js]: system should be able to launch on `env.PORT` port

File: src/routes/routes.js [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+      }
+
+      res.status(400).send('A user with this ID was not found.');
+    },
+  )
+  .put(
+    '/users/:id',
+    userValidator.query(querySchema.userInfo),
+    userValidator.params(querySchema.findUser),
+    (req, res) => {
```
Comment (MurDaD) [inline src/routes/routes.js]: You can move these functions to separate place - thus code will look more clear

File: src/routes/routes.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+      return;
+    }
+
+    res.status(400).send('No users found.');
+  })
+  .post(
+    '/users',
+    userValidator.query(querySchema.userInfo),
+    (req, res) => {
+      const { login, password, age } = req.query;
```
Comment (MurDaD) [inline src/routes/routes.js]: These fields should not be in `query` - they should be in `body`. Also, you should use `body-parser`

File: src/routes/routes.js (hunk 4 above)
Comment (jordaninthewind) [reply src/routes/routes.js]: Is this in a separate validator controller or something of that sort? What's the best way to approach this?

File: src/routes/routes.js (hunk 5 above)
Comment (jordaninthewind) [reply src/routes/routes.js]: updated

File: src/routes/routes.js (hunk 4 above)
Comment (jordaninthewind) [reply src/routes/routes.js]: updated to body, kept validator as middleware

File: src/controllers/controller.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+  getUser(id) {
+    const currentUser = this.users.find((user) => user.id === id);
+    if (!currentUser) {
+      return false;
+    }
+
+    return currentUser;
+  }
+
+  getAutoSuggestUsers(loginSubstring, limit) {
```
Comment (MurDaD) [inline src/controllers/controller.js]: Does not work. Also returns weird error
`A user with this ID was not found.`
Please check all the datapoints with postman before submitting the review

Comment (jordaninthewind) [reply src/controllers/controller.js]: I checked it multiple times and it works with the UUID, per the spec. For example, '/users/661216f9-6be5-411e-bd97-5e4578fec651' returns the User object.

Screen Shot 2020-02-03 at 11 44 43 AM

Also, the current working branch for module 2 is Review Branch #6 and this functionality has been modified on Review Branch #7 for module 3 to work with DB primary keys.

**Pull request closed without merging**